Backpack-SwiftUI/Rating/Classes/BPKRating.swift

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKRating: View {
+
+    private let accessibilityLabel: String
+    private let value: Float
+    private let ratingScale: Scale
+    private let size: Size
+    private let subtitle: String?
+    private let showScale: Bool
+    @ViewBuilder private var titleViewBuilder: () -> AnyView
+
+    public var body: some View {
+        HStack(alignment: .lastTextBaseline, spacing: BPKSpacing.md.value) {
+            HStack(alignment: .lastTextBaseline, spacing: .zero) {
+                BPKText(
+                    ratingScale.displayedValue(from: value),
+                    style: size.fontStyle.ratingValueLabelFontStyle
+                ).foregroundColor(.textPrimaryColor)
+                if showScale {
+                    BPKText(
+                        ratingScale.displayedScale(),
+                        style: size.fontStyle.ratingScaleLabelFontStyle
+                    ).foregroundColor(.textSecondaryColor)
+                }
+            }
+            if size == .large {
+                VStack(alignment: .leading, spacing: .zero) {
+                    titleViewBuilder()
+                    subtitleView
+                }
+            } else if size == .default {
+                HStack(alignment: .center, spacing: BPKSpacing.md.value) {
+                    titleViewBuilder()
+                    subtitleView
+                }
+            }
+        }
+        .accessibilityElement()
+        .accessibilityLabel(accessibilityLabel)
+    }
+
+    public init(
+        accessibilityLabel: String,
+        value: Float,
+        ratingScale: Scale = .zeroToFive,
+        size: Size = .default,
+        subtitle: String? = nil,
+        showScale: Bool = true,
+        @ViewBuilder titleViewBuilder: @escaping () -> AnyView
+    ) {
+        self.accessibilityLabel = accessibilityLabel
+        self.value = value
+        self.ratingScale = ratingScale
+        self.size = size
+        self.subtitle = subtitle
+        self.showScale = showScale
+        self.titleViewBuilder = titleViewBuilder
+    }
+
+    public init(
+        accessibilityLabel: String,
+        title: String,
+        value: Float,
+        ratingScale: Scale = .zeroToFive,
+        size: Size = .default,
+        subtitle: String? = nil,
+        showScale: Bool = true
+    ) {
+        self.accessibilityLabel = accessibilityLabel
+        self.value = value
+        self.ratingScale = ratingScale
+        self.size = size
+        self.subtitle = subtitle
+        self.showScale = showScale
+
+        self.titleViewBuilder = {
+            AnyView(
+                BPKText(title, style: size.fontStyle.titleLabelFontStyle)
+                    .foregroundColor(.textPrimaryColor)
+            )
+        }
+    }
+
+    @ViewBuilder
+    private var subtitleView: some View {
```

Comment:
When the subtitle view gets really long, it pushes the rating component to a point where it stops displaying the rating. 
Can we apply the same logic as with the title label? Where the string gets ellipsized? 

![image](https://user-images.githubusercontent.com/728889/222643439-e01b20d9-6e7f-44b3-ac56-ae0d907f59bd.png)

Reply:
Fixed in e252f2e361d17cbdb25248b21c54bc108a8d3c4c